Feature/dashboard header

tureey opened the pull request:
# Details
This PR refers to the Dashboard's Header Component

## Description
Create a Header component

## Issue
https://github.com/Coding-Coach/coding-coach/issues/255

Depends on #259 
Closes #255

tureey commented:
**I am working in progress.**

Things left:
* Wait for the components PR #259 to be merged so I can use the media included in
* Generate the translation for `Dashboard` (I executed compile and extract and still not working) @crysfel  Any idea?
* Use the same `spacing size` for the padding of the Header and the `width` not used by the notification
* Component tests

**Do you have any idea for the third point?**
_I want to add some `padding: X` to an element in both sides and to another Component (notification/card) give a `width: calc(100% - 2*(this_padding_size))` so both elements can be aligned_

emmawedekind commented:
I have messaged @steoneill regarding the status of #259. If I receive no response within a day or so I'll re-assign. Can you continue once that is merged @tureey?

crysfel commented on src/pages/dashboard/components/Header.js:
```diff
@@ -0,0 +1,44 @@
+        <span className={classNames('text-white text-2xl font-bold')}>
+          <Trans id="dashboard.header.title" />
+        </span>
+        <img src={logoSrc} className="w-12" />
```
Please use the `Image` component that we have

crysfel commented on src/pages/dashboard/components/Header.js:
```diff
@@ -0,0 +1,44 @@
+      {banner && (
+        <div
+          style={{ marginTop: '-6rem' }}
+          className={classNames('mx-auto w-5/6 p-8', 'bg-white shadow', 'md:w-3/5 xl:w-2/5')}
```
ahhhh... maybe these where in different lines? If so use just use backticks instead.

crysfel commented on src/pages/dashboard/components/IllustratedMessage.js:
```diff
@@ -0,0 +1,25 @@
+    <Fragment>
+      <img className="w-6" src={imageSrc} />
+
+      <span className={classNames('block text-secondary-dark text-lg font-light my-4')}>
```
please remove `classNames` is not needed here

crysfel replied:
you would use `classNames` when you want to dynamically set styles

crysfel commented on src/pages/dashboard/components/IllustratedMessage.js:
```diff
@@ -0,0 +1,25 @@
+function Banner({ imageSrc, message, button }) {
+  return (
+    <Fragment>
+      <img className="w-6" src={imageSrc} />
```
Use the `Image` component we have

crysfel commented on src/pages/dashboard/Dashboard.js:
```diff
@@ -0,0 +1,25 @@
+  getBanner = () => (
+    <IllustratedMessage
+      imageSrc={LogoSrc}
+      message="You have 3 new mentorship messages!"
```
i18n?

tureey replied:
Well this should not even be here. This is an example hardcoded. I guess this "banner" or "Dashboard notifications" or "dashboard messages" will be get through redux. 

I just wanted to point out how to use it.. maybe its better to do this on the story and then remove from the view for now? @crysfel

crysfel commented on src/pages/dashboard/Dashboard.js:
```diff
@@ -0,0 +1,25 @@
+      message="You have 3 new mentorship messages!"
+      button={
+        <Button onClick={() => console.log('click')} size="small">
+          open
```
i18n?

crysfel commented on src/pages/dashboard/Dashboard.stories.js:
```diff
@@ -0,0 +1,8 @@
+
+import Dashboard from './Dashboard';
+
+storiesOf('Dashboard', module).add('Header', () => {
```
I don't think we need storybook for the pages... I feel like storybook should only be for components, thoughts?

crysfel commented on src/containers/App/App.js:
```diff
@@ -2,14 +2,15 @@ import React from 'react';
       <Router>
         <Home path="/" />
         <Login path="/auth/login" />
+        <Dashboard path="/dashboard" />
```
No, reach-router support something like this:

```
<PrivateViews path="/app">
  <Dashboard path="dashboard" />
</PrivateViews>
```

Inside `PrivateViews` we are going to render the actual content or the login page, depending if the user is logged in or not. But that's not part of this task, for now just render the content inside `PrivateViews`.

tureey commented:
Why github still keep saying of conflicts? I pulled `develop`, merged into my `branch`.
I had to resolve some conflicts and now I'm updated

crysfel commented:
That's weird... did you `rebase` your branch?

tureey commented:
oh.. I did not! 😆  @crysfel

emmawedekind commented:
Any update @tureey ? :)

tureey commented:
**I think all the feedback given has been corrected. Do another check please! @crysfel **

The **only pending thing is to update the logo I'm currently using** _(the white version for the dark)_.
Still pending because https://github.com/Coding-Coach/coding-coach/pull/291 is not closed yet.

I even prepared to make the "banner/notification or whatever" optional. So if we don't pass it the header is rendered without it.

The point is to pass this Banner or the name we decide with a given props provided by redux. (This component is done, we will just need to think about this banner combinations of icon/text)

_see pictures attached and give feedback please!_

![1](https://user-images.githubusercontent.com/16169890/51630566-c6829b80-1f4a-11e9-816a-60fb885c2cfd.png)
![2](https://user-images.githubusercontent.com/16169890/51630568-c6829b80-1f4a-11e9-8bd0-2adb82b9cdaf.png)

emmawedekind commented:
Looks so great @tureey! My only critique is that we need the gradient on the background to not be so white on the right side (the logo fades into it). Do you need a gradient from me?

emmawedekind commented:
Also when the user is on a wider view port, we should make the header text larger ;)

emmawedekind commented:
> @emmawedekind I'll look for which gradient I used. But yes, you can provide me one if needed!
> 
> Related tot he text, you mean the _Dashboard_ text? or the text in the card? I'll update it tomorrow

The Dashboard text :)

crysfel commented:
Build is failing 😬

tureey commented:
Yes, I haven't invest time since I am waiting for a gradient from @emmawedekind and trent to merge the icons PR @crysfel

tureey commented:
I've updated it.

- Updated the logo. Now we use the dark one
- Changed the `IllustratedMessage` API
- Incremented sizes of text and logo

If desired by @emmawedekind still need to update the gradient.

Should we wait for it? Should we merge and open a new issue to update the gradient once she provides us it? @crysfel 

PS: The build in local is working OK for me...

crysfel commented:
Just fix the issues in travis and we should merge!

There are some linting issues:

```
./src/pages/auth/PrivateViews.js
  Line 1:  'React' is defined but never used  no-unused-vars
./src/pages/dashboard/components/Header.js
  Line 5:  'logoSrc' is defined but never used  no-unused-vars
  Line 6:  'Image' is defined but never used    no-unused-vars```

Just remove those unused variables and it should be good to go.

Thanks

tureey commented:
Tomorrow at first in the morning I'll do it. (it means in 12h) @crysfel

tureey commented:
@crysfel it's strange because the `imgSrc` and `<Image>` are used..
my build in local works. And my precommit as well.

Can you confirm it's still happening for you @crysfel ?

crysfel commented:
Mmmm... I'm not really sure, we are running the linter locally as well 🤔
